scipy/stats/tests/test_morestats.py

```diff
@@ -57,33 +58,43 @@ def test_bad_arg(self):
 
 
 class TestAnderson(TestCase):
+
     def test_normal(self):
-        rs = RandomState(1234567890)
+        rs = RandomState(12345)
         x1 = rs.standard_exponential(size=50)
         x2 = rs.standard_normal(size=50)
-        A,crit,sig = stats.anderson(x1)
+        A, crit, sig = stats.anderson(x1)
         assert_array_less(crit[:-1], A)
-        A,crit,sig = stats.anderson(x2)
+        A, crit, sig = stats.anderson(x2)
         assert_array_less(A, crit[-2:])
 
     def test_expon(self):
-        rs = RandomState(1234567890)
+        rs = RandomState(12345)
         x1 = rs.standard_exponential(size=50)
         x2 = rs.standard_normal(size=50)
-        A,crit,sig = stats.anderson(x1,'expon')
+        A, crit, sig = stats.anderson(x1, 'expon')
         assert_array_less(A, crit[-2:])
         olderr = np.seterr(all='ignore')
         try:
-            A,crit,sig = stats.anderson(x2,'expon')
+            A, crit, sig = stats.anderson(x2, 'expon')
         finally:
             np.seterr(**olderr)
         assert_(A > crit[-1])
 
+    def test_gumbel(self):
+        rs = RandomState(12345)
+        x1 = rs.gumbel(size=50)
+        A, crit, sig = stats.anderson(x1, 'gumbel_l')
+        assert_array_less(crit[:-1], A)
+        A, crit, sig = stats.anderson(x1, 'gumbel_r')
+        assert_array_less(A, crit[-2:])
```

Comment:
I think the test should follow the pattern of the other tests and sample from the Null distribution, i.e. `gumbel_r`.

AFAIR, to check that these functions are correct, I ran a small Monte Carlo. Generate 1000 samples of gumbel_r and check that the rejection rate is approximately the right one.
Those checks/tests are not in the scipy test suite.